Adding README file (#7)
Cambridge-ICCS/TCTrack: README.md

# TCTrack

The TCTrack package aims to provide functionality for running tropical cyclone (TC)
tracking algorithms and analysing the results. This can be used to compare the output
of different algorithms for a variety of data sources.

> [!WARNING]  
> The software is currently in early development and therefore subject to significant
> changes.

## Installation

### Dependencies
The package requires python3 (>=3.9).

### Package Installation
First clone the repository using `git`:
```sh
git clone https://github.com/Cambridge-ICCS/TCTrack
```

> Setup and activate a virtual environment:

Reviewer: Nit.
```suggestion
Set up and activate a virtual environment:
```

# TCTrack

> The TCTrack package provides the functionality for running tropical cyclone (TC)
> tracking algorithms and analysing the results. This can be used to compare the output
> of different algorithms for a variety of data sources.

Reviewer: ```suggestion
The TCTrack package aime to provide functionality for running tropical cyclone (TC)
tracking algorithms and analysing the results. This can be used to compare the output
of different algorithms for a variety of data sources.
```
Consider softening this statement for accuracy now?

> The software is currently in early development and therefore subject to significant

Reviewer: Consider fancy formatting with
```markdown
> [!WARNING]  
> Critical content demanding immediate user attention due to potential risks.
```

changes.

For the full documentation, see the website:
[https://tctrack.readthedocs.io](https://tctrack.readthedocs.io).

> 

Reviewer: Perhaps remove, but add an note to #5 to add back when tackled.

## Installation

### Dependencies
The package requires python3 (>=3.9).

### Package Installation
First clone the repository using `git`:
```sh
git clone https://github.com/Cambridge-ICCS/TCTrack
```

Setup and activate a virtual environment:
```sh
python3 -m venv .venv
source .venv/bin/activate
```
When finished using TCTrack this can be turned off with `deactivate`.

Then install the package using `pip`:
```sh

> pip install TCTrack

Reviewer: ```suggestion
pip install .
```

Reviewer: Consider adding a comment for developers about creating an editable install?
Perhaps under development section below.

```

## Using TCTrack
```python
import TCTrack
#[Example]
```

## Contributing

Contributions and collaborations are welcome.

For bugs, feature requests, and clear suggestions for improvement please
[open an issue](https://github.com/Cambridge-ICCS/TCTrack/issues).

If you have added something to _TCTrack_ that would be useful to others, or can
address an [open issue](https://github.com/Cambridge-ICCS/TCTrack/issues), please
[fork the repository](https://github.com/Cambridge-ICCS/TCTrack/fork) and open a
pull request.

### Additional dependencies for development

- [pytest](https://docs.pytest.org/en/stable/) is used for unit testing.
- [ruff](https://docs.astral.sh/ruff/) is used for formatting and linting.

### Testing

All code contributions should have accompanying unit tests to ensure that all parts of

> the code are functioning properly.

Reviewer: I like this sentence, so definitely keep this.

The testing framework uses `pytest` and can be found under

> [examples/test/](https://github.com/Cambridge-ICCS/TCTrack/blob/main/examples/test/).

Reviewer: Perhaps this should be removed for now as the file doesn't exist, but perhaps add a note to #3 to say that something should be included in the documentation about how to run tests when they are added.
